sdk/avalon_sdk/interfaces/event_processor.py

```diff
@@ -0,0 +1,186 @@
+# Copyright 2020 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import logging
+from abc import ABC, abstractmethod
+
+logging.basicConfig(
+    format='%(asctime)s - %(levelname)s - %(message)s', level=logging.INFO)
+
+LISTENER_SLEEP_DURATION = 2  # second
+
+class EventProcessor(ABC):
+
+    async def listener(self, event_filter, *kargs, **kwargs):
+        """
+        This function defines a listener that runs an infinite loop
+        to listen for events from an event source. It might use pass
+        on the event_filter to the blockchain specific get_events()
```

Comment:
I would reword this. Change:
`It might use pass on the event filter to ...`
to
`The optional event_filter paramter is passed to ...`